lib/components/OrderDialog/CheckoutOrder.tsx

```diff
@@ -0,0 +1,275 @@
+import { MapPin } from "lucide-react"
+import { Button } from "lib/components/ui/button"
+import { Input } from "lib/components/ui/input"
+import { cn } from "lib/utils"
+import { useState } from "react"
+import { toast } from "react-hot-toast"
+
+interface SavedAddress {
+  id: number
+  name: string
+  street: string
+  city: string
+  state: string
+  zipCode: string
+  country: string
+  isDefault: boolean
+}
+
+interface CheckoutOrderProps {
+  finalCost: number
+  onConfirmCheckout: () => void
+  onBack: () => void
+}
+
+export const CheckoutOrder = ({
+  finalCost,
+  onConfirmCheckout,
+  onBack,
+}: CheckoutOrderProps) => {
+  const [selectedAddressId, setSelectedAddressId] = useState<number | null>(
+    null,
+  )
+
+  const [savedAddresses] = useState<SavedAddress[]>([
+    {
+      id: 1,
+      name: "Home",
+      street: "123 Main St",
+      city: "San Francisco",
+      state: "CA",
+      zipCode: "94105",
+      country: "USA",
+      isDefault: true,
+    },
+    {
+      id: 2,
+      name: "Office",
+      street: "456 Market St",
+      city: "San Francisco",
+      state: "CA",
+      zipCode: "94103",
+      country: "USA",
+      isDefault: false,
+    },
+  ])
```

Comment:
just a note: there are already endpoints to load this from the registry-api on prod (maybe in the fake), so you'll want to use react-query or something to track/mutate that